src/aiida/manage/configuration/settings.py

```diff
@@ -26,45 +27,69 @@
 DEFAULT_ACCESS_CONTROL_DIR_NAME = 'access'
 
 # Assign defaults which may be overriden in set_configuration_directory() below
-AIIDA_CONFIG_FOLDER: pathlib.Path = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
-DAEMON_DIR: pathlib.Path = AIIDA_CONFIG_FOLDER / DEFAULT_DAEMON_DIR_NAME
-DAEMON_LOG_DIR: pathlib.Path = DAEMON_DIR / DEFAULT_DAEMON_LOG_DIR_NAME
-ACCESS_CONTROL_DIR: pathlib.Path = AIIDA_CONFIG_FOLDER / DEFAULT_ACCESS_CONTROL_DIR_NAME
+glb_aiida_config_folder: pathlib.Path = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
```

Comment:
Maybe I am missing something but why not keep the current name? It would be more consistent with the other global vars above, and would reduce the diff in this PR significantly.

```suggestion
AIIDA_CONFIG_FOLDER: pathlib.Path = pathlib.Path(DEFAULT_AIIDA_PATH).expanduser() / DEFAULT_CONFIG_DIR_NAME
```

Reply:
Because it is not a constant but a variable. I simply want to distinguish it from real constants above. It is also from pyright regards it as an type error.

Reply:
I don't think I agree with this, it is still a constant semantically, it's just being computed from other constants at import time.

I don't think we should change code just because (some) LSP complains. What exactly is pyright complaining about? It's certainly not a type error, perhaps we're just not adhering to pyrights naming standards?

Also I guess using pyright here is not ideal since we're using `mypy` as a type checker, so you'll be constantly fighting the differences in behaviour between them.

Reply:
But it is still not a constant but a global variable. It escape from the module scope anyway. I try to follow the strict singleton pattern in the new commit. 
The name is changed with underscore in front to be only used for the module.